### PR TITLE
Prompt to save edited tab

### LINKED ISSUE #60: No confirmation when exiting with unsaved changes
Some kind of dialog asking if the user wants to save & quit or quit without saving should probably be added. Currently, the app just closes, discarding unsaved changes:

https://github.com/pop-os/cosmic-edit/assets/7199422/5b2945f5-759c-4437-aa90-9fd1f39d1f67

#### New file:

image

#### Existing file:

image

#### New file:

image

#### Existing file:

image

### PR DESCRIPTION
**Pull request by joshuamegnauth54**
Closes #60 

I implemented a prompt for the user to handle unsaved changes when closing a tab. Currently, changes are discarded.

This works but I have to make a few changes so I'll mark this PR as a draft.

Remaining tasks:
- [x] Button to discard changes
- [x] Handle closing tabs other than the active tab

### CODE REVIEW

**Comment by joshuamegnauth54** [comment]
This only handles manually closed tabs because session support is in the works, but I can try to update the code to prompt for unsaved changes upon closing the editor too.

File: src/main.rs [hunk 1]
```diff
@@ -1008,6 +1012,60 @@ impl App {
...
+        };
+
+        // TODO: Maybe a button to show diffs in a split?
+        // And more info, such as the path if any?
+        // let diff = widget::text(fl!("diff"));
+        // let diff_button = widget::button(diff.into());
+
+        // Discards unsaved changes
+        let discard = widget::text(fl!("discard"));
+        let discard_button = widget::button(discard)
```
Comment (jackpot51) [inline src/main.rs:1050]: This should be widget::button::destructive to get the right padding.

File: src/main.rs [hunk 2]
```diff
@@ -1008,6 +1012,60 @@ impl App {
...
+            .on_press(Message::Save)
+            .style(theme::Button::Suggested)
+            .width(Length::Fill);
+
+        // "Save As" is only shown if the file has been saved previously
+        // Rationale: The user may want to save the modified buffer as a new file
+        let save_as_button = match self.tab_model.data(entity) {
+            Some(Tab::Editor(tab)) if tab.path_opt.is_some() => {
+                let save_as = widget::text(fl!("save-as"));
+                let save_as_button = widget::button(save_as)
```
Comment (jackpot51) [inline src/main.rs:1034]: This should be widget::button::suggested to get the right padding

**Review by jackpot51 — APPROVED** [review]
This looks good. There may be UI changes later

**Pull request merged**